crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/entry_point.rs

```diff
@@ -234,9 +299,23 @@ fn mocked_call_info(call: CallEntryPoint, ret_data: Vec<StarkFelt>) -> CallInfo
             failed: false,
             gas_consumed: 0,
         },
-        vm_resources: VmExecutionResources::default(),
+        resources: ExecutionResources::default(),
         inner_calls: vec![],
         storage_read_values: vec![],
         accessed_storage_keys: HashSet::new(),
     }
 }
+
+#[must_use]
+fn aggregate_nested_syscall_counters(trace: &Rc<RefCell<CallTrace>>) -> SyscallCounter {
+    let mut result = SyscallCounter::new();
+    for nested_call in &trace.borrow().nested_calls {
+        let nested_call = nested_call.borrow();
+        result = sum_syscall_counters(result, &nested_call.used_syscalls);
+        for sub_trace in &nested_call.nested_calls {
+            let sub_trace_counter = aggregate_nested_syscall_counters(sub_trace);
+            result = sum_syscall_counters(result, &sub_trace_counter);
+        }
+    }
+    result
```

Comment:
I don't understand the logic here, why are 2 loops needed?
Wouldn't something like this be enough?

```rust
fn aggregate_nested_syscall_counters(trace: &Rc<RefCell<CallTrace>>) -> SyscallCounter {
    let mut result = SyscallCounter::new();
    for nested_call in &trace.borrow().nested_calls {
        let nested_call = nested_call.borrow();
        let sub_trace_counter = aggregate_nested_syscall_counters(nested_call);
        result = sum_syscall_counters(result, &sub_trace_counter);
    }
    result
}
```

Reply:
It doesn't take into account the top level syscalls

Reply:
```rust
let mut result = trace.borrow().used_syscalls;
```
does this not?

Reply:
I mean the original function doesn't do it - on purpose, but yours does

Reply:
Still, the logic seems off to me - why recursively call `aggregate_nested_syscall_counters` from sub_trace?

If it's equivalent, I would rather prefer this:
```rust
fn aggregate_nested_syscall_counters(trace: &Rc<RefCell<CallTrace>>) -> SyscallCounter {
    let mut result = SyscallCounter::new();
    for nested_call in &trace.borrow().nested_calls {
        let sub_trace_counter = aggregate_syscall_counters(nested_call);
        result = sum_syscall_counters(result, &sub_trace_counter);
    }
    result
}

fn aggregate_syscall_counters(trace: &Rc<RefCell<CallTrace>>) -> SyscallCounter {
    let mut result = trace.borrow().used_syscalls;
    for nested_call in &trace.borrow().nested_calls {
        let sub_trace_counter = aggregate_nested_syscall_counters(nested_call);
        result = sum_syscall_counters(result, &sub_trace_counter);
    }
    result
}
```
or
```rust
fn aggregate_nested_syscall_counters(trace: &Rc<RefCell<CallTrace>>) -> SyscallCounter {
    aggregate_syscall_counters(trace) - trace.borrow().used_syscalls
}
```

Reply:
done